report-viewer/src/components/FilesContainer.vue

```diff
@@ -13,7 +13,9 @@
         :lines="!files.get(file)?.lines ? [] : files.get(file)?.lines"
         :matches="!matches.get(file) ? [] : matches.get(file)"
         :panel-id="containerId"
-        :title="file.length>40?'..'+file.substring(file.length-40,file.length):file"
+        :title="convertSubmissionIdToName(file, submissionId).length > 40?
```

Comment:
I would suggest moving this to an additional function that calls `convertSubmissionIdToName` with the length check of 40 and so on, to minimize the complexity of this section